lib/file-helper.js

```diff
@@ -45,6 +47,117 @@ exports.getNunjucksCode = path => {
   return content
 }
 
+// This helper function takes a path of a macro options file and
+// returns the options data grouped by tables to output in markup
+exports.getMacroOptions = componentName => {
+  let options = []
+  let processedOptions = []
+  let primaryTable = {
+    'name': 'Primary options',
+    'id': 'primary',
+    'options': []
+  }
+  processedOptions.push(primaryTable)
+
+  if (componentName === 'text-input') {
+    componentName = 'input'
+  }
+
+  try {
+    let optionsFilePath = path.join(paths.govukfrontendcomponents, componentName, 'macro-options.json')
+    options = JSON.parse(fs.readFileSync(optionsFilePath, 'utf8'))
+  } catch (err) {
+    console.error(err)
+    process.exit(1) // Exit with a failure mode
+  }
+
+  for (let option of options) {
+    // Example of an option
+    // {
+    //    name: "errorMessage",
+    //    type: "string",
+    //    required: false,
+    //    description: "Options for the errorMessage component"
+    //    isComponent: true
+    // }
+    if (option.isComponent) {
+      // Create separate table data for components that are hidden in the
+      // Design System
+      if (option.name === 'hint' || option.name === 'label') {
+        let otherComponentOptions
+        try {
+          let otherComponentPath = path.join(paths.govukfrontendcomponents, option.name, 'macro-options.json')
+          otherComponentOptions = JSON.parse(fs.readFileSync(otherComponentPath, 'utf8'))
+        } catch (err) {
+          console.error(err)
+          process.exit(1) // Exit with a failure mode
+        }
+
+        if (otherComponentOptions) {
+          processedOptions.push({
+            'name': 'Options for ' + option.name,
+            'id': option.name,
+            'options': otherComponentOptions
+          })
+          option.description += ` See [${option.name}](#nunjucks-options-table-${option.name}).`
+        }
+      // Otherwise just link to that component
+      } else {
+        let optionName = (option.name).replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase() // camelCase into kebab-case
+        let otherComponentPath = '/components/' + optionName + '/#options-example-default'
```

Comment:
Sometimes the `default` example is not the first example on the page

(Non blocking, as discussed in person)